### ISSUE TITLE
Settlement service returns 400 (bad request) when settlements or windows do not exist

### ISSUE THREAD

**Issue by partiallyordered** [mojaloop/project#2344]
**Summary**:
Calling `GET /v2/settlements` when there are no settlements returns
```
400 Bad Request
{
  "errorInformation": {
    "errorCode": "3100",
    "errorDescription": "Generic validation error - Settlements not found"
  }
}
```
Calling `GET /v2/settlementWindows` with a query, for example `GET /v2/settlementWindows?fromDateTime=2021-06-29T23:00:00.000Z&toDateTime=2021-06-30T22:59:59.999Z`, when there are no settlement windows returns:
```
400 Bad Request
{
  "errorInformation": {
    "errorCode": "3100",
    "errorDescription": "Generic validation error - settlementWindow by filters: {fromDateTime:2021-06-29T23:00:00.000Z,toDateTime:2021-06-30T22:59:59.999Z} not found"
  }
}
```

I would characterise this response in English as:
> Caller misused the API: I don't have any settlements
which is unconventional and confusing.

I have not examined all endpoints on the settlement service, there may be others returning similarly unconventional responses.

The consequence of these responses will vary for different users of the API. Some users will experience an exception using the default configuration of their API client. Others will have to implement special handling for this response. Both will likely need to test for this specific response text.

**Severity**:
(Low, Medium, High)

**Priority**:
(Critical, Medium, Low)

**Expected Behavior**
My opinion is that these endpoints should return 200 with the empty list when there are no settlements or windows. This means many users will not need to implement exception handling, nobody will need to test for the specific response text, and many users will not need to implement special handling at all, as their code may be implemented to operate in an equally valid manner on a return value of an empty list.

In other words: for a minor change in implementation in the settlement service, we can free all users of the settlement service of the burdens of
1. interpreting a confusing, unconventional response
2. implementing special handling of this response

**Steps to Reproduce**
As above

**Specifications**
  - Component (if known): central settlement
  - Version: latest (and IIRC as long as these endpoints have existed)
  - Platform:
  - Subsystem:
  - Type of testing:
  - Bug found/raised by:

**Notes**:

- Severity when opened:
- Priority when opened:

**Comment by bushjames** [comment]
:+1:

**Comment by elnyry-sam-k** [comment]
@partiallyordered - thanks for logging the issue. What would you mark the severity, priority of this issue as?

**Comment by partiallyordered** [comment]
I wasn't really sure. Perhaps some guidance on this in the issue template would be useful. As you're asking, I guess what's intended is for the author of the issue to determine severity and priority in their own context. I've updated the issue to reflect these values for my context. (Low, low).

I'm happy to do some work to fix this, just wanted agreement that it is indeed an issue before I committed any effort to it. (In fact, I already committed some effort to it, but the code doesn't run with node 16, so I opted to raise this issue before going down that rabbit hole :grin: ).

**Comment by elnyry-sam-k** [comment]
Thanks Matt.. Yes it looks like an issue

Yes it is from the perspective / context of whoever is raising the bug; the bug triage team can then update from a more high-level perspective..

if you can raise a PR with your fix, that would be great! Thanks in advance

**Comment by elnyry-sam-k** [comment]
Reviewed in PI-27, marked for further analysis.
